modules/search-api/src/main/scala/io/renku/search/api/HttpApplication.scala

```diff
@@ -21,89 +21,26 @@ package io.renku.search.api
 import cats.effect.{Async, Resource}
 import cats.syntax.all.*
 import fs2.io.net.Network
-import io.circe.syntax.given
-import io.renku.search.http.borer.TapirBorerJson
 import io.renku.solr.client.SolrConfig
 import org.http4s.dsl.Http4sDsl
 import org.http4s.server.Router
 import org.http4s.{HttpApp, HttpRoutes, Response}
-import sttp.apispec.openapi.Server
-import sttp.apispec.openapi.circe.given
-import sttp.tapir.*
-import sttp.tapir.docs.openapi.OpenAPIDocsInterpreter
-import sttp.tapir.server.ServerEndpoint
-import sttp.tapir.server.http4s.Http4sServerInterpreter
-import io.renku.search.query.Query
-import io.renku.search.query.docs.SearchQueryManual
+import io.renku.search.api.routes.*
 
 object HttpApplication:
   def apply[F[_]: Async: Network](
       solrConfig: SolrConfig
   ): Resource[F, HttpApp[F]] =
     SearchApi[F](solrConfig).map(new HttpApplication[F](_).router)
 
-class HttpApplication[F[_]: Async](searchApi: SearchApi[F])
-    extends Http4sDsl[F]
-    with TapirBorerJson
-    with TapirCodecs:
+final class HttpApplication[F[_]: Async](searchApi: SearchApi[F]) extends Http4sDsl[F]:
 
-  private val businessRoot = "search"
+  private val search = new SearchRoutes[F](searchApi)
+
+  private val businessEndpoints = search.endpoints
 
   lazy val router: HttpApp[F] =
     Router[F](
-      s"/$businessRoot" -> businessRoutes,
-      "/" -> operationsRoutes
+      "/search" -> (OpenApiRoute(businessEndpoints) <+> search.routes),
```

Comment:
I was wondering if it wouldn't be good to keep the `/search` segment somewhere so it can be shared between this class and the `SearchRoutes` (the definition of server for the OpenAPI spec).

Reply:
Ah, I removed it because it wasn't used anywhere else - but I now saw the other occurrence in `OpenApiRoute`. Thanks, will bring it back!